### DISCUSSION TITLE
Support Git LFS tracking by file size

### DISCUSSION THREAD

**Discussion by wilg** [community/community#16689] [Other Feature Feedback, Questions, & Ideas]
Git LFS is required by GitHub at a certain point. For the better part of a decade the top request here has been to allow tracking files by size. This has been repeatedly ignored by GitHub employees in the LFS issue tracker. Posting here for visibility. Please see the original discussion here: https://github.com/git-lfs/git-lfs/issues/282

**Comment by oleg620** [comment]
Additionally, we need different thresholds for binary and textual files.
Preferably the system will identify which files are binary without requiring the user to specify it by file extension.

**Comment by samjacoby** [comment]
This is a major issue for us; we just want to opt-in all files once they hit GitHub's required 100mb threshold (or more likely a much lower limit to speed up clones — e.g. 5Mb). It would simplify our GitHub and git-lfs integrations if we could just start tracking files once they hit GitHub's threshold. 
```
git lfs track > 100Mb
```

**Comment by wilg** [comment]
Just FYI, I emailed GitHub support to ask them to take a look at implementing this. I tried my best to make sure the product team there understood the issue and advocate that it was something GitHub should put resources into handling. I'm not sure they actually understand the use case, nor am I sure they view LFS as a part of their product that they are responsible for improving. So, seems like they aren't interested in lifting much of a finger on it. 

Seems like the only hope at this point is for someone from the community to help out with https://github.com/git-lfs/git-lfs/issues/282. Fortunately, some community people seem to be taking up the task.

I'm pretty disappointed in the GitHub team and the particular maintainer on the LFS repo, and feel pretty beat up for trying to help, so I'm out.
